src/Nest/CommonAbstractions/Infer/ForecastIds/ForecastIds.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Elasticsearch.Net;
+
+namespace Nest
+{
+	[DebuggerDisplay("{DebugDisplay,nq}")]
+	public class ForecastIds : IUrlParameter, IEquatable<ForecastIds>
+	{
+		private readonly List<string> _forecastIds;
+
+		public ForecastIds(IEnumerable<string> forecastIds) => _forecastIds = forecastIds?.ToList() ?? new List<string>();
+
+		public ForecastIds(string forecastIds) => _forecastIds = forecastIds.IsNullOrEmpty()
+			? new List<string>()
+			: forecastIds.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries)
+				.Select(s => s.Trim())
+				.ToList();
+
+		internal IReadOnlyList<string> Ids => _forecastIds;
+
+		private string DebugDisplay => ((IUrlParameter)this).GetString(null);
+
+		public bool Equals(ForecastIds other)
+		{
+			if (Ids == null && other.Ids == null) return true;
+			if (Ids == null || other.Ids == null) return false;
+
+			return Ids.Count == other.Ids.Count && !Ids.Except(other.Ids).Any();
+		}
+
+		string IUrlParameter.GetString(IConnectionConfigurationValues settings) => string.Join(",", _forecastIds);
+
+		public static implicit operator ForecastIds(string forecastIds) =>
+			forecastIds.IsNullOrEmptyCommaSeparatedList(out var list) ? null : new ForecastIds(list);
+
+		public static implicit operator ForecastIds(string[] forecastIds) => forecastIds.IsEmpty() ? null : new ForecastIds(forecastIds);
```

Comment:
This should have the same semantics as using the constructor. With an empty `forecastIds`, this returns `null`, but the constructor assigns a `new List<string>()` to the backing field. Should both ways of constructing an instance of `ForecastIds` result in the same outcome?